docs/src/components/icons-list.astro

```diff
@@ -9,15 +9,40 @@ const icons = Object.keys(Icons) as (keyof typeof Icons)[];
 	{
 		icons.map((icon) => {
 			return (
-				<div class="icon-preview">
+				<div class="icon-preview" id={icon}>
 					<Icon name={icon} size="1.5rem" />
 					<span>{icon}</span>
+					<span class="copied">copied!</span>
 				</div>
 			);
 		})
 	}
 </div>
 
+<script>
+  const icons = document.querySelectorAll('.icon-preview');
+  icons.forEach((icon) => {
+		let intervalId;
+    icon.addEventListener('click', () => {
+			const iconName = icon.id;
+			const copiedValue = `<Icon name="${iconName}" size="2rem" />`;
+			navigator.clipboard.writeText(copiedValue);
+			
+			const iconText = icon.children[1];
+			const confirm = icon.children[2];
+			confirm?.classList.toggle('copied');
+			iconText?.classList.toggle('icon-name');
+
+			intervalId = setTimeout(() => {
+				confirm?.classList.toggle('copied');
+				iconText?.classList.toggle('icon-name');
+			}, 1000)
+			
+    });
+		clearInterval(intervalId);
+  });
```

Comment:
To work with our new approach, we have to change the script to instead of toggling between two spans, use only one and change its `textContent` property through the click event.

I've also made a few additional changes:
1. To be more expressive and clear, instead of getting the icon's text through the children array, I used a querySelector based on the icon's id and span element.
2. The `clearInterval` function doesn't work with `setTimeout`, for that we'd use `clearTimeout`, and with our current implementation, we don't need to use it.
```suggestion
  const icons = document.querySelectorAll('.icon-preview');
  icons.forEach((icon) => {
    icon.addEventListener('click', () => {
      const iconName = icon.id;
      const copiedValue = `<Icon name="${iconName}" size="2rem" />`;
      navigator.clipboard.writeText(copiedValue);

      const iconLabel = document.querySelector(`#${iconName} span`);
      iconLabel!.textContent = 'Copied!';

      setTimeout(() => {
        iconLabel!.textContent= iconName;
      }, 1000);
    });
  });
```